src/decoder.rs

```diff
@@ -58,7 +60,7 @@ pub struct Decoder<R> {
 
     restart_interval: u16,
     color_transform: Option<AdobeColorTransform>,
-    is_jfif: bool,
+    jfif_app0: Option<JfifApp0>,
```

Comment:
Are `color_transform` and `jfif_app0` mutually exclusive? If so, perhaps this could hold an AppData or Option<AppData> instead?